mxnet-bot/LabelBotFullFunctionality/LabelBot.py

```diff
@@ -0,0 +1,247 @@
+        :return Fully converted string
+        """
+        cleans = re.sub("[^0-9a-zA-Z]", sub_string, raw_string)
+        return cleans.lower()
+
+    def find_all_labels(self):
+        """
+        This method finds all existing labels in the repo
+        :return A set of all labels which have been extracted from the repo
+        """
+        url = 'https://api.github.com/repos/{}/{}'.format(self.repo, "labels")
+        response = requests.get(url, auth=self.auth)
+        response.raise_for_status()
+
+        # Getting total pages of labels present
+        if "link" not in response.headers:
+            pages = 1
+        else:
+            pages = int(self.clean_string(response.headers['link'], " ").split()[-3])
+
+        all_labels = []
+        for page in range(1, pages + 1):
+            url = 'https://api.github.com/repos/' + self.repo + '/labels?page=' + str(page) \
+                  + '&per_page={}'.format(self.LABEL_PAGE_PARSE, repo=self.repo)
+            response = requests.get(url, auth=self.auth)
+            for item in response.json():
+                all_labels.append(item['name'].lower())
+        self.all_labels = set(all_labels)
+        return set(all_labels)
+
+    def format_labels(self, labels):
+        """
+        This method formats labels that a user specifies for a specific issue. This is meant
+        to provide functionality for the operations on labels
+        :param labels: The messy labels inputted by the user which we want to format
+        :return: Formatted labels to send for CRUD operations
+        """
+        assert self.all_labels, "Find all labels first"
+        # clean labels, remove duplicated spaces. ex: "hello  world" -> "hello world"
+        labels = [" ".join(label.split()) for label in labels]
+        labels = [label for label in labels if label.lower() in self.all_labels]
+        return labels
+
+    def add_github_labels(self, issue_num, labels):
+        """
+        This method is to add a list of labels to one issue.
+        It checks whether labels exist in the repo, and adds existing labels to the issue
+        :param issue_num: The specific issue number we want to label
+        :param labels: The labels which we want to add
+        :return Response denoting success or failure for logging purposes
+        """
+        labels = self.format_labels(labels)
+        issue_labels_url = 'https://api.github.com/repos/{repo}/issues/{id}/labels' \
+            .format(repo=self.repo, id=issue_num)
+
+        response = requests.post(issue_labels_url, json.dumps(labels), auth=self.auth)
+        if response.status_code == 200:
+            logging.info('Successfully added labels to {}: {}.'.format(str(issue_num), str(labels)))
+            return "Added labels successfully"
+        else:
+            logging.error("Could not add the labels")
+            logging.error(response.json())
+            return "Failed to add labels"
```

Comment:
Why are we returning response?
We should either log or return values - not both. In this case as we are logging, returning is not needed. Please remove.